identity_iota_core/src/client/identity_client.rs

```diff
@@ -149,7 +150,11 @@ pub trait IotaIdentityClientExt: IotaIdentityClient {
     let (_, alias_output) = self.get_alias_output(id).await?;
 
     let document: &[u8] = alias_output.state_metadata();
-    IotaDocument::unpack(did, document, true)
+    let mut resolved_document: IotaDocument = IotaDocument::unpack(did, document, true)?;
+
+    set_controller_and_governor_addresses(&alias_output, &mut resolved_document, &self.network_name().await?);
```

Comment:
The controller fields are only set when calling `resolve_did` here, we should set them in `unpack_from_block` too (which is called in `publish_did_output`). 
We may need an `unpack_from_output`/`unpack_from_alias_output` and try centralise the logic there?